torchvision/prototype/transforms/_utils.py

```diff
@@ -31,15 +33,13 @@ def _check_fill_arg(fill: Optional[Union[FillType, Dict[Type, FillType]]]) -> No
             raise TypeError("Got inappropriate fill arg")
 
 
-def _setup_fill_arg(
-    fill: Optional[Union[FillType, Dict[Type, FillType]]]
-) -> Union[Dict[Type, FillType], Dict[Type, None]]:
+def _setup_fill_arg(fill: Union[FillType, Dict[Type, FillType]]) -> Dict[Type, FillType]:
     _check_fill_arg(fill)
 
     if isinstance(fill, dict):
         return fill
 
-    return defaultdict(lambda: fill)  # type: ignore[return-value]
+    return defaultdict(lambda: fill)  # type: ignore[return-value, arg-type]
```

Comment:
If we don't do the `arg-type` ignore we get:
```
torchvision/prototype/transforms/_utils.py:42: error: Argument 1 to
"defaultdict" has incompatible type
"Callable[[], Union[Union[int, float, Sequence[int], Sequence[float], None], Dict[Type[Any], Union[int, float, Sequence[int], Sequence[float], None]]]]";
expected "Optional[Callable[[], Union[float, Sequence[float], None]]]" 
[arg-type]
        return defaultdict(lambda: fill)  # type: ignore[return-value]
                           ^
torchvision/prototype/transforms/_utils.py:42: note: Error code "arg-type" not covered by "type: ignore" comment
Found 1 error in 1 file (checked 222 source files)
```

Reply:
I think this is worth investigating. At some point `mypy` thinks that fill can also be an `int`. Although that is part of the correct type, it shouldn't be for the actual type. I guess we forgot to update something. @datumbox I can look into this if you want.

Reply:
I agree. I've also felt that this was worth noting. I have no idea if this is a failure or mypy or it's seeing something I dont. If you have any input, I would love to get an advice.

Reply:
This is a false alarm by `mypy` right before the line `mypy` thinks that `fill` has type

```
Union[builtins.float, typing.Sequence[builtins.float], None]
```

Note that for `mypy` `float == Union[int, float]` or to illustrate this more: for `mypy` the following is true while this will fail for Python `isinstance(0, float)`. I'm not sure why they went for this, but generic numbers support is sketchy in general. This is why you don't see the `int` and `Sequence[int]` here because for `mypy` they are subtypes of `float` and `Sequence[float]` and therefore redundant.

However, at the time we arrive at the line, `mypy` suddenly thinks that `fill` can be a `Dict` again. I can't find the issue now, but this is longstanding. For some reason `mypy` doesn't forward the more concrete type to inner functions like this `lambda`.

Both

```suggestion
    return defaultdict(lambda: cast(FillType, fill))
```

and

```suggestion
    return defaultdict(cast(Callable[[], FillType], lambda: fill))
```

will eliminate both previously ignored errors, but will raise a new `redundant-cast` error :roll_eyes: 

Thus, we have to ignore one either way and not having `cast` in there makes it more readable.